libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -190,6 +193,11 @@ ModuleEntitlements getEntitlementsOrThrow(Class<?> callerClass, Module requestin
             }
         }
 
+        if (requestingModule.isNamed() == false) {
+            // agents are the only thing running non-modular
```

Comment:
How is that enforced in the system? Since this is a security feature, we should probably explain why we think this is safe.